app/generators/questionnaire_generator.rb

```diff
@@ -385,13 +402,39 @@ def update_id(id, sub_id)
       "#{start}_#{sub_id}_#{endd}".to_sym
     end
 
+    def update_ids(ids, sub_id)
+      result = []
+      ids.each do |id|
+        result << id # keep the original id
+        result << update_id(id, sub_id)
+      end
+      result
+    end
+
+    def update_options(current_options, sub_id)
+      current_options.map do |optorig|
+        option = optorig.clone
+        if option.is_a?(Hash)
+          option[:hides_questions] = update_ids(option[:hides_questions], sub_id) if option[:hides_questions].present?
```

Comment:
je zou die `if option[:..].present?` ook in de `update_ids` functie zetten. Dat is consistent met `update_id`..

Reply:
Dat codepad in `update_id` wordt in principe nooit gebruikt omdat id altijd een waarde heeft. De reden dat ik die present? er nog achter heb staan is omdat ik vind dat als er geen hides_questions (of shows_questions) entry in die hash bestaat, dan moet je m ook niet gaan aanmaken. In principe zou je ook de hele option array mee kunnen geven en daar dan naar toe kunnen schrijven in update_ids, maar dat vind ik minder transparant.